code/modules/atmospherics/gasmixtures/gas_types.dm

```diff
@@ -12,7 +12,14 @@ GLOBAL_LIST_INIT(hardcoded_gases, list("o2","n2","co2","plasma")) //the main fou
 		if(initial(gas.moles_visible) != null)
 			gas_info[META_GAS_OVERLAY] = new /obj/effect/overlay/gas(initial(gas.gas_overlay))
 		gas_info[META_GAS_DANGER] = initial(gas.dangerous)
-		.[initial(gas.id)] = gas_info
+		gas_info[META_GAS_ID] = initial(gas.id)
+		.[gas_path] = gas_info
+
+/proc/gas_id2path(id)
+	var/meta_gas = meta_gas_list()
+	for(var/path in meta_gas)
+		if(meta_gas[path][META_GAS_ID] == id)
+			return path
```

Comment:
this should probably write a stack_trace so that we can find out where the old syntax is being used

Reply:
or maybe in assert_gas where its used

Reply:
The old syntax is still supported, namely for gas strings. This is because you can't have embedded expressions in constant strings (including var declarations or map modifications) and I would much rather support simple IDs than long paths if we can't even get the benefits of the paths for these particular cases.